src/main/kotlin/io/titandata/titan/providers/local/Status.kt

```diff
@@ -5,22 +5,40 @@
 package io.titandata.titan.providers.local
 
 import io.titandata.titan.providers.Container
+import io.titandata.client.apis.RepositoriesApi
+import java.text.DecimalFormat
+import kotlin.math.log10
 
 class Status (
-    private val getContainersStatus: () -> List<Container>
+    private val getContainersStatus: () -> List<Container>,
+    private val repositoriesApi: RepositoriesApi = RepositoriesApi()
 ){
+    private val n = System.lineSeparator()
+
+    /**
+     * https://stackoverflow.com/a/5599842
```

Comment:
FYI, you may want to compare the output of this with RsyncExecutor::numberToString() in titan-server. The latter is based on the ZFS implementation, which pays special attention to behavior near boundaries (e.g. 1023MiB instead of 0.99GiB).